### PR TITLE
feat: render content in the Static Legend

### LINKED ISSUE #512: Graph Legend: render content - add series labels and values
__Proposal:__
Have an option to add the series labels to a chart when it is showing multiple series.

__Current behavior:__
When a chart has multiple series at the moment, the only way to work out what plot line a series relates to is by hovering the mouse over the series.  

__Desired behavior:__
Instead of hovering over the chart, have an option to show the series names and what colour plot line they relate to so you can identify see a specific series.

__Alternatives considered:__
None available

__Use case:__
When showing a dashboard on a monitor which is displayed, say, in a command center, it is not practical to keep moving the mouse over a plotline to see what series it relates to, so having the labels annotated underneath the chart would make this much easier to use.

### PR DESCRIPTION
**Pull request by TCL735**
Closes #512 

Screen Shot 2021-04-21 at 8 01 19 PM

### CODE REVIEW

**Review by hoorayimhelping — APPROVED** [review]
looks fine to me, but it's kind of hard to feel confident checking over such a large change. I have a feeling we'll miss some stuff

File: giraffe/src/components/LegendStyles.ts [hunk 1]
```diff
@@ -205,25 +205,25 @@ const tooltipDotsColumnStyles = (
...
       verticalAlign: 'middle',
-      padding: tooltipTablePadding,
+      padding: legendTablePadding,
     }
   }
 
   const colors =
-    tooltipData.length && Array.isArray(tooltipData[0].colors)
-      ? tooltipData[0].colors
+    legendData.length && Array.isArray(legendData[0].colors)
```
[the remark this answers is not in the document]
Comment (TCL735) [reply giraffe/src/components/LegendStyles.ts:220]: the expression before `&&` already checks for null. We probably could just combine into one but this change was about the name and i didn't really look at making it updating it

File: giraffe/src/components/StaticLegendBox.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+interface LegendOptionsOverridden {
+  config: SizedConfig
+  staticLegend: StaticLegend
+}
+const overrideLegendConfig = (
+  config: SizedConfig,
+  staticLegend: StaticLegend
+): LegendOptionsOverridden => {
+  const configWithDefaults = Object.assign(
```
[the remark this answers is not in the document]
Comment (TCL735) [reply giraffe/src/components/StaticLegendBox.tsx:31]: Yeah it does. Will change it. But it should be the other way around. The defaults come first and then get overridden.

File: giraffe/src/components/StaticLegendBox.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  const {
+    legendBackgroundColor: backgroundColor,
+    legendBorder: border,
+    legendFont: font,
+    legendFontBrightColor: fontBrightColor,
+  } = configOverride
+  return (
+    <div
+      className="giraffe-static-legend"
+      style={{
```
[the remark this answers is not in the document]
Comment (TCL735) [reply giraffe/src/components/StaticLegendBox.tsx:79]: We can. But this is not like Clockface where almost everything has a className or dataTestId as a prop. The only consumer of this name would probably be a unit test for the component which we don't have. There isn't complicated logic in the component itself. The conversion util which makes the data has a unit test.

It's missing the data-test-id at the very least, so i will add that.

**Comment by TCL735** [comment]
I did a force push to fix merge conflicts after Zoe's changes to upgrade storybook, which required re-working some of the knob functions' passed-in arguments

File: giraffe/src/utils/legend/staticLegend.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  const valueColumn = {
+    key: valueColumnKey,
+    name: `Latest ${valueColumnKey}`,
+    type: spec.table.getColumnType(valueColumnKey),
+    colors,
+    values,
+  }
+
+  const legendColumns = columnKeys.map(key => {
+    // TODO: should be using formatter from config
```
Comment (jrenee42) [inline giraffe/src/utils/legend/staticLegend.ts:53]: aren't we already using the formatter? (in the tooltip itself).

so this todo should be defunct
(please verify, and if so remove)

**Review by jrenee42 — CHANGES_REQUESTED** [review]

File: giraffe/src/utils/legend/staticLegend.ts (hunk 4 above)
Comment (TCL735) [reply giraffe/src/utils/legend/staticLegend.ts:53]: it's a valid todo which we will address in the follow-up PRs for stacked lines and band chart

**Pull request merged**